packages/office-ui-fabric-react/src/components/pickers/BasePicker.types.ts

```diff
@@ -188,92 +188,27 @@ export interface IBasePickerProps<T> extends React.Props<any> {
   theme?: ITheme;
 }
 
-export interface IBasePickerSuggestionsProps {
-  /**
-   * Function that specifies what to render when no results are found.
-   */
-  onRenderNoResultFound?: IRenderFunction<void>;
-
-  /**
-   * The text that should appear at the top of the suggestion box.
-   */
-  suggestionsHeaderText?: string;
-
-  /**
-   * The text that should appear at the top of the most recently used box.
-   */
-  mostRecentlyUsedHeaderText?: string;
-
-  /**
-   * The text that should appear when no results are returned.
-   */
-  noResultsFoundText?: string;
-
-  /**
-   * Suggestions root className.
-   */
-  className?: string;
-
-  /**
-   * Suggestions List className.
-   */
-  suggestionsClassName?: string;
-
-  /**
-   * ClassName for suggestion items.
-   */
-  suggestionsItemClassName?: string;
-
-  /**
-   * The text that should appear on the button to search for more.
-   */
-  searchForMoreText?: string;
-
-  /**
-   * The text that appears indicating to the use to force resolve the input
-   */
-  forceResolveText?: string;
-
-  /**
-   * The text to display while the results are loading.
-   */
-  loadingText?: string;
-
-  /**
-   * The text to display while searching for more results in a limited suggestions list.
-   */
-  searchingText?: string;
-
-  /**
-   * A renderer that adds an element at the end of the suggestions list if it has more items than resultsMaximumNumber.
-   */
-  resultsFooterFull?: () => JSX.Element;
-
-  /**
-   * A renderer that adds an element at the end of the suggestions list when there are fewer than resultsMaximumNumber.
-   */
-  resultsFooter?: () => JSX.Element;
-
-  /**
-   * Maximum number of suggestions to show in the full suggestion list.
-   */
-  resultsMaximumNumber?: number;
-
-  /**
-   * Indicates whether to show a button with each suggestion to remove that suggestion.
-   */
-  showRemoveButtons?: boolean;
-
-  /**
-   * Screen reader message to read when there are suggestions available.
-   */
-  suggestionsAvailableAlertText?: string;
-
-  /**
-   * An ARIA label for the container that is the parent of the suggestions.
-   */
-  suggestionsContainerAriaLabel?: string;
-}
+export interface IBasePickerSuggestionsProps<T = any>
```

Comment:
I don't think `T` is actually used for any of the picked props, is it? If not, I think it should just be removed here.

Reply:
oh actually it is required.

Like i mentioned in the description, `'resultsFooterFull'` and `'resultsFooter'` are narrowed to not take the suggestion prop of the item into account when rendering. (hence diverging from the picked type).

Reply:
If we were to redefine the type as 
```typescript
/**
 * Subset of picker options that may be legally passed through a picker to its
 * internal Suggestions component.
 */
export interface IBasePickerSuggestionsProps<T = any>
  extends Pick<
    ISuggestionsProps<T>,
    | 'onRenderNoResultFound'
    | 'suggestionsHeaderText'
    | 'mostRecentlyUsedHeaderText'
    | 'noResultsFoundText'
    | 'className'
    | 'suggestionsClassName'
    | 'suggestionsItemClassName'
    | 'searchForMoreText'
    | 'forceResolveText'
    | 'loadingText'
    | 'searchingText'
    | 'resultsMaximumNumber'
    | 'showRemoveButtons'
    | 'suggestionsAvailableAlertText'
    | 'suggestionsContainerAriaLabel'
  > {
  /**
   * A renderer that adds an element at the end of the suggestions list it has more items than resultsMaximumNumber.
   */
  resultsFooterFull?: () => JSX.Element;

  /**
   * A renderer that adds an element at the end of the suggestions list it has fewer items than resultsMaximumNumber.
   */
  resultsFooter?: () => JSX.Element;
}
```

Then there would be no problem

Reply:
(But this isn't 100% reflective of the actual requirements of the component)